nexus/db-queries/src/db/datastore/instance.rs

```diff
@@ -204,9 +207,27 @@ impl From<InstanceAndActiveVmm> for external::Instance {
                 .hostname
                 .parse()
                 .expect("found invalid hostname in the database"),
+
+            auto_restart_cooldown: value
+                .instance()
+                .auto_restart
+                .cooldown
+                // Eventually, this may have a project-level default, so if the
+                // instance doesn't provide a value we'll have to use the
+                // project's default if one exists. For now, though, fall back
+                // to the hard- coded default if the instance hasn't overridden
+                // it.
+                .unwrap_or(InstanceAutoRestartConfig::DEFAULT_COOLDOWN)
```

Comment:
So, this is a bit tricky. The reason I wanted to add this value to the external API instance type in the first place is because --- as @davepacheco pointed out in https://github.com/oxidecomputer/omicron/pull/6503#discussion_r1757360480 --- if we want to have long cooldown periods, like on the order of an hour, it's really important to be able to show the user "this instance hasn't been restarted because it's still waiting out the cooldown period", so they don't see an instance that's been `Failed` for a long time and go, "wait, why is this still Failed?". Therefore, in order to show "this instance can't be restarted until...", the UI needs to know both the last restart time *and* the cooldown period, so I wanted the API to always return whatever cooldown is being applied, regardless of whether it came from an explicit configuration or the defaults.

I get the concern about not having a GET and then a PUT set values that weren't previously set, though. The other option that occurred to me is returning `NULL`s in the API when the default is in use and then have some new API for getting the default values, which the console could use if an instance has nulls for these fields. The reason I didn't do that here is because that felt like something we would need to do as part of #1015 (project-level defaults), and I didn't really want to throw together an ad-hoc version of that API in this PR...

If we think it would be better to do it that way, though, I'm happy to make this return `None`s when the default configuration is in use and wait until work on #1015 adds a first-class API for getting default values. But, that makes implementing (at least some of) #1015 more urgent, because I would really prefer not to ship a long default cooldown time without exposing cooling-down instances to the user...

What do you think?

Reply:
Okay, https://github.com/oxidecomputer/omicron/pull/6503/commits/37fa32707fd9b6fb77dde04fe7ac321c1d66e743 resolves this by changing the `Instance` view to represent the cooldown period as the *time at which the cooldown will expire* rather than the start time and duration, which I think should make it much more obvious that it's not a config value to be clobbered by a potential future PUT API.